### PR TITLE
introducing some basic tests for the rInterface

### PR DESCRIPTION
**Pull request by Sparrow0hawk**
This is some old work that should now pass the test suite after we updated `rpy2`. 

It aims to introduce some minimal tests for the rInterface with some dummy data. It crucially alters parts of the r_interface API by not hard-setting the working directory but assuming the code is run from the top level directory and just hardcoding path locations to the R scripts and `R/py_int` directory.

It also adds a step to the testing workflow to install the rampuaR package.

### CODE REVIEW

File: .github/workflows/pythontesting.yml [hunk 1]
```diff
@@ -47,7 +52,7 @@ jobs:
       shell: bash -l {0}
       run: |
         conda activate ramp-ua
-        pytest --cov=microsim tests/ --cov-report=xml
+        pytest --full-trace --cov=microsim tests/ --cov-report=xml
```
Comment (Sparrow0hawk) [inline .github/workflows/pythontesting.yml:55]: This needs switching back
```suggestion
        pytest --cov=microsim tests/ --cov-report=xml
```

**Comment by Sparrow0hawk** [comment]
@nickmalleson can you check this PR please by doing the following on your local machine:
- [ ] - Running the test suite with `pytest`
- [ ]  - Running the python/R model for an acceptable number of iterations to ensure it runs

Steps to checkout this specific PR:

```bash
$ cd RAMP-UA 

$ git fetch origin pr/276/head:pr-276

$ git checkout pr-276
```

**Comment by Sparrow0hawk** [comment]
@nickmalleson When you get a spare moment can you give this a go so we can merge this

**Comment by nickmalleson** [comment]
Sorry @Sparrow0hawk I missed this. 

Have tried the following:

```
$ cd RAMP-UA 
$ git fetch origin pr/276/head:pr-276
$ git checkout pr-276
```

but have a difficulty with the fetch:

```
$ git fetch origin pr/276/head:pr-276
fatal: couldn't find remote ref pr/276/head
```

Any ideas?

**Comment by Sparrow0hawk** [comment]
@nickmalleson apologies my error it should be:
```bash
$ git fetch origin pull/276/head:pr-276
```

Will correct above.

**Comment by nickmalleson** [comment]
All good @Sparrow0hawk , 59 tests passed

**Review by nickmalleson — APPROVED** [review]

**Pull request merged**